runscanner/src/main/java/uk/ac/bbsrc/tgac/miso/runscanner/Scheduler.java

```diff
@@ -0,0 +1,222 @@
+  }
+
+  public boolean isScanningNow() {
+    return scanningNow;
+  }
+
+  @Value("${runscanner.configFile}")
+  public void setConfigurationFile(String filename) {
+    ObjectMapper mapper = new ObjectMapper();
+    try {
+      roots = Arrays.stream(mapper.readValue(new File(filename), SuppliedDirectoryConfig[].class)).map(source -> {
+        Configuration destination = new Configuration();
+        destination.setPath(new File(source.getPath()));
+        destination.setTimeZone(TimeZone.getTimeZone(source.getTimeZone()));
+        for (RunProcessor processor : RunProcessor.INSTANCES) {
+          if (processor.getName().equals(source.getName()) && processor.getPlatformType() == source.getPlatformType()) {
+            destination.setProcessor(processor);
+            break;
+          }
+        }
+        return destination;
+      }).collect(Collectors.toConcurrentMap(Configuration::getPath, Function.identity()));
+    } catch (IOException e) {
+      log.error("Confguration is bad.", e);
+    }
+  }
+
+  public synchronized void start() {
+    if (scanDirectoriesFuture == null) {
+      scanDirectoriesFuture = scheduler.scheduleAtFixedRate(() -> {
+        if (scanningNow) {
+          reentered.inc();
+          return;
+        }
+        scanningNow = true;
+        long startTime = System.nanoTime();
+        try (StreamCountSpy<Pair<File, Configuration>> newRuns = new StreamCountSpy<>(newRunsScanned);
+            StreamCountSpy<File> attempted = new StreamCountSpy<>(attemptedDirectories);
+            StreamCountSpy<File> accepted = new StreamCountSpy<>(acceptedDirectories)) {
+          roots.entrySet().stream().filter(entry -> entry.getValue().isValid()).flatMap(entry -> {
+            return Arrays.stream(entry.getKey().listFiles(File::isDirectory)).peek(attempted)
+                .filter(directory -> directory.canRead() && directory.canExecute())
+                .peek(accepted).filter(directory -> !workToDo.contains(directory) && !processing.contains(directory)
+                    && !failed.contains(directory) && !finishedWork.containsKey(directory))
+                .map(directory -> new Pair<>(directory, entry.getValue()));
+          }).peek(newRuns).forEach(entry -> {
+            final File directory = entry.getKey();
+            final Supplier<NotificationDto> process = entry.getValue().getProcessor().apply(directory, entry.getValue().getTimeZone());
+            workToDo.add(directory);
+            workPool.submit(() -> {
+              processing.add(directory);
+              workToDo.remove(directory);
+              NotificationDto result = process.get();
+              if (result == null) {
+                failed.add(directory);
+              } else {
+                finishedWork.put(directory, result);
+              }
+              processing.remove(directory);
+            });
+          });
+        } catch (Exception e) {
+          log.error("Error scanning directory.", e);
```

Comment:
This is at the instruments level, not the run level. If an instrument directory is bad, we should ignore and continue and let monitoring pick it up. We'll need to have an alert that if `attempted > accepted`, some instrument is dead.